revolution/motor.py

```diff
@@ -124,3 +128,46 @@ def _revolution(self) -> None:
             with self.environment.contexts() as contexts:
                 contexts.motor_revolution_period = revolution_period
                 contexts.motor_speed = motor_speed
+
+    def cruise_control(self) -> None:
+        with self.environment.contexts() as contexts:
```

Comment:
Here, when you enter the context with the ``with`` statement, you are acquiring the lock associated with the context. Since we effectively have an infinite loop, this will deadlock attempts to access the contexts from other threads. You must move the above statement so that it only contains interaction with contexts (line 169, 171). My suggestion is that in those lines, you don't set to contexts, but just define as ``motor...input = ...`` and later acquire the context in Line 172 and set the context attributes.